20230127 johani

johanix opened the pull request:
This is now (I think) "feature complete". But I'm sure there are all sorts of horrible language issues, as I wrote this in one go.

I make the pull request mostly so that you'll get the notification, not that I think this should be merged exactly as it is. More like, "subject for discussion".

Regards,
Johan

johanix committed:
* fixed a bunch of non-ASCI stuff (mostly double quote and single quote)

* started to present the case for adding a scheme parameter. Not complete.

johanix committed:
* rototilling of most of the document to update to a general use
  of new RR NOTIFY everywhere and removing the SRV alternative.

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -210,9 +210,10 @@ the registry itself, it is expected that in the ICANN RRR model, some
 For such parents, the receiving service should notify the appropriate
 registrar, over any communication channel deemed suitable between
-registry and registry (such as EPP).
+registry and registry (such as EPP, or possibly by forwarding the
```
`registry and registrar`

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -239,26 +240,27 @@ the notification is the parent, to improve the speed and efficiency of
-notification to the parent) is not sufficient then a more general
+However, there may exist cases where this scheme (sending the
+notification to the parent) is not sufficient and a more general
 design is needed. However, it is strongly desireable that the child is
```
This and the previous sentence begin with `However`

johanix replied:
Rewrote that a little bit.

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -239,26 +240,27 @@ the notification is the parent, to improve the speed and efficiency of
 where the only scheme defined here is scheme=1 with the interpretation
-above.
+that when a new CDS (or CDNSKEY or CSYNC) is published, a NOTIFY(CDS)
+or NOTIFY(CSYNC) should be sent to the address and port listed in the
```
We run the risk of confusing the reading by overloading the meaning of `NOTIFY`, which can now be either the type or the message. To remedy this, we should say `a NOTIFY(CDS) or NOTIFY(CSYNC) *message*`

johanix replied:
I agree.

johanix replied:
I added a short "Terminology" paragraph just after the Introduction to further clarify this.

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -389,9 +386,69 @@ records is ensured.
+        The domain name of the target host providing the service of
+        listening for generalised notifications of the specified type.
+        There MUST be one or more address records for this name, the
+        name MUST NOT be an alias (in the sense of RFC 1034 or RFC
```
Why not an alias, and why so strong (MUST NOT)? -- Aliases seem fine to me.

johanix replied:
I stole that text from the RFC for SRV and didn't really think it through. I removed that sentence, as I agree that a CNAME should not be a problem.

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -389,9 +386,69 @@ records is ensured.
+        name MUST NOT be an alias (in the sense of RFC 1034 or RFC
+        2181). 
+
+        A Target of "." means that the service is decidedly not
+        available at this domain.
```
Is that different from not having the record at all?

johanix replied:
Hmm. While this is also stolen from RFC 2782 I did think about that. My thought was that it made sense to clearly signal the intention NOT to provide a target address for notifications, as a sort of "negative response" that clearly signals that there is no point in looking elsewhere. 

But on further reflection I think it is better to keep it simple so I removed this sentence also.

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -400,12 +457,11 @@ case of the more "horizontal" NOTIFY(DNSKEY), however, the
-solution to all the new types of notification signaling. Eg.:
+A new record type would therefore make it possible to more easily
+associate the special processing needed with the new type rather than
+the standard SRV RRtype that occur in completely other scenarios than
```
`occurs`

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -400,12 +457,11 @@ case of the more "horizontal" NOTIFY(DNSKEY), however, the
+A new record type would therefore make it possible to more easily
+associate the special processing needed with the new type rather than
+the standard SRV RRtype that occur in completely other scenarios than
+are described here. type would provide a cleaner solution to all the
```
`type` --> `The NOTIFY type`?

johanix replied:
Right, trivial errors all over.

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -470,11 +531,11 @@ Per {{!RFC8552}}, IANA is requested to create the "Generalized DNS
+| CSYNC       |      1 | parent.  | [this document] |
+| DNSKEY      |      1 | zone.    | [this document] |
 
 [TODO: (1) Detail out use cases? (2) Add entries via RFC or web form?]
```
Should be expert review

johanix replied:
Hmm? I don't understand your comment here?

peterthomassen replied:
I commented on my own old TODO in line 540, which raises the question by what mechanism entries can later be added to the NOTIFY IANA registry, such as via an RFC, or via a simple web form.

It then occurred to me that we should say that it can be done based via Expert Review, which is one of the "accredited" mechanisms. 

If you agree, we can borrow language [here](https://www.ietf.org/archive/id/draft-ietf-dnsop-dns-catalog-zones-09.html#name-iana-considerations-2).

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -389,9 +386,69 @@ records is ensured.
+        Class.
+
+RRtype
+        The type of generalised NOTIFY that this NOTIFY RR defines
+		the desired target address for. Currently only the types CDS,
+        CSYNC and DNSKEY are suggested to be supported, but there is
+		no protocol issue should a use case for additional types of
+		notifications arise in the future.
+ 		
+Scheme
+	    The scheme for locating the desired notification address.
+        The range is 0-255. This is a 16 bit unsigned integer in
+		network byte order. The value 0 is an error. The value 1 is
+		described in this document and all other values are currently
+		unspecified.
+
+Port
+	    The port on the target host of the notification service. The
+        range is 0-65535. This is a 16 bit unsigned integer in network
+		byte order.
```
inconsistent indentation

peterthomassen reviewed (changes requested):
Very cool! Thank you, great progress. Once this is polished and merged, I think we can submit the new revision.

Some comments/nits in the code. Otherwise, I noticed we have a split between British and American English, most significantly with the word "generali[sz]ed". I suggest using "z" because it's also in the draft name.

I believe the rendering job failed due to linting errors, such as inconsistent indentation. The job output suggest to "Run 'make fix-lint' to automatically fix some errors".

johanix committed:
* fixed all Peter's comments but the last one that I don't understand :-)

peterthomassen commented on draft-thomassen-dnsop-generalized-dns-notify.md:
```diff
@@ -408,32 +436,27 @@ Class
-
-        A Target of "." means that the service is decidedly not
-        available at this domain.
+        There MUST be one or more address records for this name. 
```
This wording also may be read to preclude aliases. Suggestion: `This name MUST resolve to one or more address records.`

johanix replied:
Hmm. I see your point and agree with your suggestion.

peterthomassen replied:
done

peterthomassen reviewed (changes requested):
Thanks! Agree with all your changes.

Modulo
- the one comment I added in this review,
- some whitespace stuff,
- the Expert Review thingy,

I think we're done for an updated submission.

1.) Do you agree?
2.) Depending on your editor, sorting out the whitespace stuff that the linter expects may be tedious. It's easy for me to do; if you prefer I can take that on.
3.) I can take on the Expert Review thingy.

johanix committed:
* more tweaks. I think I've addressed all your comments except
  possibly the white space issue. Please take a look, and if there's
  still white space problems then please do whatever you want to
  fix them.

johanix committed:
* typo

johanix committed:
foo

peterthomassen committed:
fix whitespace

peterthomassen committed:
some more clarifications

peterthomassen commented:
I fixed the whitespace issue (which fixed the rendering job), and added some clarifications where I had trouble understanding. I think this is now ready for merging and draft submission. What do you think?
